cmd/soroban-cli/src/commands/network/container/start.rs

```diff
@@ -51,137 +55,152 @@ pub struct Cmd {
+        self.print_stop_message();
+        Ok(())
+    }
+
+    fn get_image_name(&self) -> String {
+        // this can be overriden with the `-t` flag
+        let mut image_tag = match &self.network {
+            Network::Pubnet => "latest",
+            Network::Futurenet => "future",
+            _ => "testing", // default to testing for local and testnet
+        };
+
+        if let Some(image_override) = &self.image_tag_override {
+            println!(
+                "Overriding docker image tag to use '{image_override}' instead of '{image_tag}'"
+            );
+            image_tag = image_override;
         }
-    );
 
-    println!("{stop_message}");
-    Ok(())
-}
+        format!("{DOCKER_IMAGE}:{image_tag}")
+    }
 
-fn get_container_args(cmd: &Cmd) -> Vec<String> {
-    [
-        format!("--{}", cmd.network),
-        "--enable rpc,horizon".to_string(),
-        get_protocol_version_arg(cmd),
-        get_limits_arg(cmd),
-    ]
-    .iter()
-    .filter(|&s| !s.is_empty())
-    .cloned()
-    .collect()
-}
+    fn get_container_args(&self) -> Vec<String> {
+        [
+            format!("--{}", self.network),
+            "--enable rpc,horizon".to_string(),
+            self.get_protocol_version_arg(),
+            self.get_limits_arg(),
+        ]
+        .iter()
+        .filter(|&s| !s.is_empty())
+        .cloned()
+        .collect()
+    }
+
+    // The port mapping in the bollard crate is formatted differently than the docker CLI. In the docker CLI, we usually specify exposed ports as `-p  HOST_PORT:CONTAINER_PORT`. But with the bollard crate, it is expecting the port mapping to be a map of the container port (with the protocol) to the host port.
+    fn get_port_mapping(&self) -> HashMap<String, Option<Vec<PortBinding>>> {
+        let mut port_mapping_hash = HashMap::new();
+        for port_mapping in &self.ports_mapping {
+            let ports_vec: Vec<&str> = port_mapping.split(':').collect();
+            let from_port = ports_vec[0];
+            let to_port = ports_vec[1];
+
+            port_mapping_hash.insert(
+                format!("{to_port}/tcp"),
+                Some(vec![PortBinding {
+                    host_ip: None,
+                    host_port: Some(from_port.to_string()),
+                }]),
+            );
+        }
 
-fn get_image_name(cmd: &Cmd) -> String {
-    // this can be overriden with the `-t` flag
-    let mut image_tag = match cmd.network {
-        Network::Pubnet => "latest",
-        Network::Futurenet => "future",
-        _ => "testing", // default to testing for local and testnet
-    };
-
-    if let Some(image_override) = &cmd.image_tag_override {
-        println!("Overriding docker image tag to use '{image_override}' instead of '{image_tag}'");
-        image_tag = image_override;
+        port_mapping_hash
     }
 
-    format!("{DOCKER_IMAGE}:{image_tag}")
-}
+    fn container_name(&self) -> Name {
+        Name::new(self.name.clone(), Some(self.network))
```

Comment:
This seems like the only place where it could be one or the other.
```suggestion
        Name::new(self.name.cloned().unwrap_or(self.network))
```
This would simplify all the other places where you are currently requiring `None`, and the constructor wouldn't need to return an error no have any other checks.

Reply:
Nice call - that simplifies things a bunch!